sdk/core/core-rest-pipeline/src/fetchHttpClient.ts

```diff
@@ -84,7 +84,10 @@ async function makeRequest(request: PipelineRequest): Promise<PipelineResponse>
      * of request options.
      * It will not work as you expect.
      */
-    const response = await fetch(request.url, requestInit);
+    const response = await fetch(request.url, {
+      ...requestInit,
+      ...request.requestOverrides,
```

Comment:
Nit, you might want to do `...request.requestOverrides || {}` here if the requestOverrides is undefined.

Reply:
Looks that spread undefined into an object is valid: 

>\> let a = { a: 1 }
undefined
\> let b = undefined
undefined
\> let c = { ...a, ...b }
undefined
\> c
{ a: 1 }